src/documents/admin.py

```diff
@@ -138,12 +142,25 @@ class NotesAdmin(GuardedModelAdmin):
     list_display = ("user", "created", "note", "document")
     list_filter = ("created", "user")
     list_display_links = ("created",)
+    raw_id_fields = ("document",)
+    search_fields = ("document__title",)
+
+    def get_queryset(self, request):
+        return (
+            super()
+            .get_queryset(request)
+            .select_related("user", "document__correspondent")
+        )
 
 
 class ShareLinksAdmin(GuardedModelAdmin):
     list_display = ("created", "expiration", "document")
     list_filter = ("created", "expiration", "owner")
     list_display_links = ("created",)
+    raw_id_fields = ("document",)
+
+    def get_queryset(self, request):
+        return super().get_queryset(request).select_related("document__correspondent")
```

Comment:
Why does `select_related("document__correspondent")` help here since I dont think ShareLinks should access the correspondent at all, no?